native/src/main/cpp/common.cpp

```diff
@@ -0,0 +1,42 @@
+#include "common.h"
+
+#include <cstring>
+#include <cassert>
+
+int get_enum_value(JNIEnv *env, jclass clazz, jobject enumObj) {
+    if (enumObj != NULL) {
+        jfieldID attrId = env->GetFieldID(clazz, "value", "I");
+        return env->GetIntField(enumObj, attrId);
+    }
+    return 0;
+}
+
+jobject get_object_field(JNIEnv *env, jclass clazz, jobject obj, const char* attr_name, const char* attr_class_name) {
+    jfieldID attrId = env->GetFieldID(clazz, attr_name, attr_class_name);
+    assert(attrId != NULL);
+    return env->GetObjectField(obj, attrId);
+}
+
+int get_int_field_value(JNIEnv *env, jclass clazz, jobject obj, const char* attr_name) {
+    jfieldID attrId = env->GetFieldID(clazz, attr_name, "I");
+    assert(attrId != NULL);
+    return env->GetIntField(obj, attrId);
+}
+
+int get_long_field_value(JNIEnv *env, jclass clazz, jobject obj, const char* attr_name) {
+    jfieldID attrId = env->GetFieldID(clazz, attr_name, "J");
+    assert(attrId != NULL);
+    return env->GetLongField(obj, attrId);
+}
```

Comment:
Oh yes it's a typo (should be ```long get_long_field_value(JNIEnv *env, jclass clazz, jobject obj, const char* attr_name)```).
Ok i'll refactor code with helper functions with signed and unsigned values checking as you correctly suggested :+1:

Reply:
Actually i've removed ```get_long_field_value``` because it's not used anywhere.
I've created as you suggested the functions
```
int get_int_field_value(...) // from java int
unsigned int get_uint_field_value(...) // from java int
long long get_llong_field_value(...) // from java long
unsigned long long get_ullong_field_value(...) // from java long
```